### PR TITLE
Closes #14 Moderation report api

### LINKED ISSUE #14: ModerationReport
https://github.com/FAForever/faf-java-api/issues/234

### PR DESCRIPTION
**Pull request by bukajsytlos**
Closes #14 Moderation report api

### CODE REVIEW

**Comment by axel1200** [comment]
Well it misses all the security stuff no?

**Comment by 1-alex98** [comment]
Can you also change Ban Info to have a relationship to report as requested in the document and implemted in the db?

**Review by micheljung — APPROVED** [review]

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/ModerationReport.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Type("moderationReport")
+@Getter
+@Setter
+public class ModerationReport extends AbstractEntity {
+  private String reportDescription;
+  private ModerationReportStatus reportStatus = ModerationReportStatus.AWAITING;
```
Comment (1-alex98) [inline faf-commons-api/src/main/java/com/faforever/commons/api/dto/ModerationReport.java]: Do you really wanna set default I doubt that, if you wanna make a patch then you want it to be null needing it to set to null on very patch is painfull. I would rather not have a default

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/ModerationReport.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  private String gameIncidentTimecode;
+  private String moderatorNotice;
+  private String moderatorPrivateNote;
+  @Relationship("bans")
+  private Collection<BanInfo> bans;
```
Comment (1-alex98) [inline faf-commons-api/src/main/java/com/faforever/commons/api/dto/ModerationReport.java]: BanInfo links ModerationReport and ModerationReport links BanInfo. Just saying, but you will notice soon enough you will need to make one relationShip manged and the other one not.

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/Player.java [hunk 3]
```diff
@@ -39,6 +40,12 @@
     @JsonIgnore
     private List<AvatarAssignment> avatarAssignments;
 
+    @Relationship("reporterOnModerationReports")
+    private Set<ModerationReport> reporterOnModerationReports;
```
Comment (1-alex98) [inline faf-commons-api/src/main/java/com/faforever/commons/api/dto/Player.java]: Same as above. But you will notice as soons as you modify modor. You can change it then I guess. Btw you can also use none develop commits to use modor with, you dont need to wait for this getting merged into develop ;)

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

**Review by 1-alex98 — APPROVED** [review]

**Review by Brutus5000 — APPROVED** [review]

**Pull request merged**